R/symptome_compartment_class.R

```diff
@@ -0,0 +1,226 @@
+#'                                 |            |                           |
+#'                                 -- (ihr)---> HA --(nu.hsa)----------------
+
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Class definitions
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+#' @describeIn model_symptome to set up dependent classes for model_symptome
+setClass("ggplot")
+setClass("gg")
+
+#' An S4 class of an basic SEIR model with compartments of different symptomes
+#'
+#' @slot name A string gives the name of the model
+#' @slot initial_population A list of the initial population size of each group
+#' @slot parameters A list of the model parameters
+#' @slot output A dataframe holding the ode simulation output, with columns as population groups and rows as simulation time points
+#' @slot plot_output A ggplot holding the plot of the ode simulation output
+#'
+#' @import deSolve
+#' @import ggplot2
+#' @import reshape2
+#' @import plyr
+#' @import magrittr
+
+setClass(Class = "model_symptome",
+         slots = c(
+           name = "character",
+           initial_population = "list",
+           parameters = "list",
+           output = "list",
+           plot_output = "gg"
+         ),
+         prototype = list(
+           name = NA_character_,
+           initial_population = vector(mode = "list", length = 6) %>%
+             setNames(list("S", "E", "I", "HA", "R", "CM")),
+           parameters = vector(mode = "list", length = 6) %>%
+             setNames(list("lam", "gamma", "omega", "ihr", "nu", "pdeath")),
+           output = data.frame(),
+           plot_output = ggplot()
+         ))
+              
+#' @describeIn model_symptome set initial population sizes (in fraction) and parameters
+#'
+#' @param object An object of class model_symptome
+#' @param S Numeric, initial population of the susceptible
+#' @param E Numeric, initial population of the exposed
+#' @param I Numeric, initial population of the infected with mild/no symptom
+#' @param HA Numeric, initial population of the infected with severe symptoms
+#' that need further hospitalization
+#' @param CM Numeric, initial cumulative mortality
+#' @param R Numeric, initial population of the recovered
+#' @param lam Numeric, rate at which an infected individual exposes susceptible
+#' @param gamma Numeric, rate at which exposed individuals become infected
+#' @param omega Numeric, rate at which recovered individuals become susceptible
+#' @param ihr Numeric, fraction of infected individuals develop severe symptoms
+#' @param nu Numeric vector, rate at which each infected group recover
+#' @param pdeath Numeric vector, rate of disease-caused mortality of each infected group
+#' @return An object of class model_symptome with initial population and parameters
+#'
+setGeneric(name = "set_init",
+           def = function(object, S=NA_real_, E=NA_real_, I=NA_real_,
```

Comment:
I would separate out the transmission parameters (lam, gamma etc.) from the initial compartment values (S_0, E_0 etc.) since they're two different types of object.